### PR TITLE
[FLINK-1694] [gelly] added IterationConfiguration as a way to configure ...

### PR DESCRIPTION
**Pull request by vasia**
...a VertexCentricIteration and avoid the split between create and run methods.

### CODE REVIEW

**Comment by vasia** [comment]
Hi,
any opinions on this? 
I would like to merge it soon because it changes the api and vertex-centric related issues depend on it.
Thanks!

File: docs/gelly_guide.md [hunk 1]
```diff
@@ -343,29 +343,86 @@ Vertex-centric Iterations
...
+### Configuring a Vertex-Centric Iteration
+A vertex-centric iteration can be configured using an `IterationConfiguration` object.
+Currently, the following parameters can be specified:
+
+* <strong>Name</strong>: The name for the vertex-centric iteration. The name is displayed in logs and messages 
+and can be specified using the `setName()` method.
+
+* <strong>Parallelism</strong>: The parallelism for the iteration. It can be set using the `setParallelism()` method.	
+
+* <strong>Solution set in unmanaged memory</strong>: Defines whether the solution set is kept in managed memory (Flink's internal way of keeping object in serialized form) or as a simple object map. By default, the solution set runs in managed memory. This property can be set using the `setSolutionSetUnmanagedMemory()` method.
```
Comment (fhueske) [inline docs/gelly_guide.md]: Typo: "... of keeping objectS in ..."

File: flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/Graph.java [hunk 2]
```diff
@@ -1118,30 +1119,52 @@ public boolean filter(Edge<K, EV> edge) {
...
+	 * after maximumNumberOfIterations.
 	 */
-	public <M> VertexCentricIteration<K, VV, M, EV> createVertexCentricIteration(
+	public <M> Graph<K, VV, EV> runVertexCentricIteration(
 			VertexUpdateFunction<K, VV, M> vertexUpdateFunction,
 			MessagingFunction<K, VV, M, EV> messagingFunction,
 			int maximumNumberOfIterations) {
-		return VertexCentricIteration.withEdges(edges, vertexUpdateFunction,
-				messagingFunction, maximumNumberOfIterations);
+
```
Comment (fhueske) [inline flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/Graph.java]: Function call could be forwarded to the configured `runVertexCentricIteration` to reduce code duplication.

**Comment by fhueske** [comment]
LGTM
I browsed the diffs and didn't find anything crucial. Just a typo and one possible simplification.

The change makes vertex iterations much easier to use as indicated by the implemented algorithms.

**Comment by StephanEwen** [comment]
I like the change as well! Good to go!

**Comment by vasia** [comment]
Thanks so much for the review @fhueske, @StephanEwen!
I'll make the suggested changes and merge :-)

**Pull request closed without merging**